packages/asset/contracts/AssetCreate.sol

```diff
@@ -0,0 +1,340 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/EIP712Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "./libraries/TokenIdUtils.sol";
+import "./AuthValidator.sol";
+import "./ERC2771Handler.sol";
+import "./interfaces/IAsset.sol";
+import "./interfaces/ICatalyst.sol";
+import "./interfaces/IAssetCreate.sol";
+
+/// @title AssetCreate
+/// @author The Sandbox
+/// @notice User-facing contract for creating new assets
+contract AssetCreate is
+    IAssetCreate,
+    Initializable,
+    ERC2771Handler,
+    EIP712Upgradeable,
+    AccessControlUpgradeable
+{
+    using TokenIdUtils for uint256;
+
+    IAsset private assetContract;
+    ICatalyst private catalystContract;
+    AuthValidator private authValidator;
+
+    // mapping of creator address to creator nonce, a nonce is incremented every time a creator mints a new token
+    mapping(address => uint16) public creatorNonces;
+
+    bytes32 public constant SPECIAL_MINTER_ROLE =
+        keccak256("SPECIAL_MINTER_ROLE");
+    bytes32 public constant BRIDGE_MINTER_ROLE =
```

Comment:
role not needed, you can put `asset.mint(...` in the bridge contract and delete `createBridgedAsset` function entirely

Reply:
Removed that one, thats a good suggestion.